Tchap/Modules/Registration/Form/RegistrationFormViewModel.swift

```diff
@@ -22,10 +22,24 @@ final class RegistrationFormViewModel: RegistrationFormViewModelType {
     // MARK: - Properties
     
     let loginTextViewModel: FormTextViewModelType
-    let passwordTextViewModel: FormTextViewModelType
+    var passwordTextViewModel: FormTextViewModelType
     let confirmPasswordTextViewModel: FormTextViewModelType
     let termsCheckBoxModel: FormCheckBoxModelType
     
+    // MARK: Public
+    
+    weak var delegate: RegistrationFormViewModelDelegate? {
+        didSet {
+            self.passwordTextViewModel.valueDidUpdate = { [weak self] (_, hasBeenAutoFilled) in
```

Comment:
Check if `valueDidUpdate` is nil to avoid potential allocation of this block several times.